src/arcade/patch/agent/cell/PatchCell.java

```diff
@@ -300,20 +313,45 @@ public void setState(CellState state) {
 
         switch ((State) state) {
             case PROLIFERATIVE:
-                module = new PatchModuleProliferation(this);
-                break;
+                // Throw an exception or handle the case if you accidentally use this method
+                throw new IllegalArgumentException("PROLIFERATIVE state requires simulation time.");
             case MIGRATORY:
                 module = new PatchModuleMigration(this);
                 break;
             case APOPTOTIC:
                 module = new PatchModuleApoptosis(this);
                 break;
             default:
-                module = null;
+                setModule(null);
                 break;
         }
     }
 
+    /**
+     * Sets the cell state to proliferative.
+     *
+     * @param proliferativeState the target cell state
+     * @param currentSimTime the current simulation time
+     */
+    public void setState(CellState proliferativeState, double currentSimTime) {
```

Comment:
I don't love this implementation, setState shouldn't change for one state/module. Is this only possible way to get the currentTime?